core/systems/bluetooth/bluetooth_manager.gd

```diff
@@ -0,0 +1,204 @@
+extends NodeThread
```

Comment:
`BluetoothManager` should probably be a resource so it can be referenced from anywhere.